src/configuration/CertificateDetailsModal.tsx

```diff
@@ -0,0 +1,358 @@
+import {
+  CodeBlock,
+  CodeBlockCode,
+  Modal,
+  Split,
+  SplitItem,
+  Stack,
+  StackItem,
+} from '@patternfly/react-core';
+import { pki, util } from 'node-forge';
+import { FC } from 'react';
+
+interface CertificateDetailsModalProps {
+  isModalOpen: boolean;
+  certs: pki.Certificate[];
+  secretName: string;
+  pem: string;
+  onCloseModal: () => void;
+}
+
+const CertificateDetailsModal: FC<CertificateDetailsModalProps> = ({
+  isModalOpen,
+  certs,
+  secretName,
+  pem,
+  onCloseModal,
+}) => {
+  const formatCertIssuer = (crt: pki.Certificate) => {
+    let result = '';
+    for (let i = 0; i < crt.issuer.attributes.length; i++) {
+      if (crt.issuer.attributes[i].shortName) {
+        result += crt.issuer.attributes[i].shortName;
+        result += ' = ';
+        result += crt.issuer.attributes[i].value;
+        if (i !== crt.issuer.attributes.length - 1) {
+          result += ', ';
+        }
+      }
+    }
+    return result;
+  };
```

Comment:
In JS we tend to use the foreach syntax when available:

```typescript
  const formatCertIssuer = (crt: pki.Certificate) => {
    let result = '';
    crt.issuer.attributes.forEach((attribute, i, all) => {
      result += attribute.shortName;
      result += ' = ';
      result += attribute.value;
      if (i !== all.length - 1) {
        result += ', ';
      }
    });
    return result;
  };
```

One main advantage is that you don't have to "dig" in for the `i`th attribute in the array yourself.